### PR TITLE
Changes have been successfully made in graph.py

### PR DESCRIPTION
**Pull request by Deeptig9138**
File Path Handling: Instead of os.path.dirname(__file__), I used os.getcwd() to get the current working directory, which is more suitable for interactive environments like Jupyter notebooks.
Graceful Error Handling (FileNotFoundError):A try-except block is used around the file reading process to handle situations where the file doesn't exist. It provides a user-friendly error message and avoids crashing.
Clipboard Copying Error Handling: Added error handling for missing clipboard utilities, such as FileNotFoundError for missing xclip or clip.exe.Also handles cases where the clipboard command fails with subprocess.CalledProcessError.
Clarity (Docstrings and Constants): Added docstrings to the display and copy_to_clipboard functions to explain their purpose.
Used a constant (STASH_DIR) for the directory name to make the code more readable and easy to modify in the future.

### CODE REVIEW

**Comment by Deeptig9138** [comment]
Done

**Comment by ShashwatAgrawal20** [comment]
@Deeptig9138 please link the issue you're resolving

https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

**Comment by jindalpriyanshu101** [comment]
@Deeptig9138 requested changes, and also resolve merge conflicts.

**Comment by jindalpriyanshu101** [comment]
@Deeptig9138 Can you please implement the requested changes? functions `display()` and `copy_to_clipboard` shouldn't have header comments.

**Comment by Deeptig9138** [comment]
All done

File: .deepsource.toml [hunk 1]
```diff
@@ -4,7 +4,4 @@ version = 1
 name = "python"
 
   [analyzers.meta]
-  runtime_version = "3.x.x"
-
-[[transformers]]
-name = "black"
```
Comment (this-is-yaash) [inline .deepsource.toml:10]: I especially want the **`black`** linter to be used in this project. I believe we can keep these lines.

File: package/graph.py [hunk 2]
```diff
@@ -1,38 +1,65 @@
...
+                    input=text.strip().encode(),
+                    check=True,
+                )
+            else:
+                raise OSError("Unsupported operating system")
+
+        except FileNotFoundError:
+            print("Error: Clipboard utility not found. Cannot copy to clipboard.")
+        except subprocess.CalledProcessError as e:
+            print(f"Error during clipboard copying: {e}")
```
Comment (this-is-yaash) [inline package/graph.py:51]: We don't need obvious comments that would expose the integration process.
Instead, create print statements that resemble the error typically seen in python.

File: package/graph.py [hunk 3]
```diff
@@ -1,38 +1,65 @@
...
+def display(snippet_name, source_code):
+    """
+    Display the snippet's content and copy it to the clipboard.
 
-    with open(snippet_path, "r") as file:
-        source_code = file.read()
+    Args:
+        snippet_name (str): The name of the snippet.
+        source_code (str): The source code to be copied and displayed.
+    """
```
Comment (this-is-yaash) [inline package/graph.py:12]: uncomment!

File: package/graph.py [hunk 4]
```diff
@@ -1,38 +1,65 @@
...
-                check=True,
-            )
-        else:
-            raise OSError("Unsupported operating system")
+        """
+        Copy the given text to the system clipboard.
+
+        Args:
+            text (str): Text to be copied to the clipboard.
+        """
```
Comment (this-is-yaash) [inline package/graph.py:20]: uncomment!

File: package/graph.py [hunk 5]
```diff
@@ -1,38 +1,65 @@
...
+    # Display the snippet's content
+    print(f"Snippet Name: {snippet_name}\n")
+    print(source_code)
+
+
+# Example usage
+if __name__ == "__main__":
+    example_name = "Sample Code Snippet"
+    example_code = """print("Hello, World!")"""
+    display(example_name, example_code)
```
Comment (this-is-yaash) [inline package/graph.py:65]: We don't need these lines!
I'm confused, could you briefly explain why they were added?

**Review by this-is-yaash — COMMENTED** [review]
Could you please make the necessary changes as per the review request?

**Comment by Wahid7852** [comment]
Hey Deepti, sorry for the trouble but it would be really appreciated if you close this PR and create a fresh PR with these changes into a single commit?

Thanks!

**Comment by Wahid7852** [comment]
Hey Deepti, sorry if I wasn't clear, I meant to close this PR and create a new PR with the same changes.
You can do this by getting a clean fork of the repo again
doing the necessary changes and then PR them.

Why? This will make the commit history clean.

**Comment by this-is-yaash** [comment]
Hey @Deeptig9138, make sure to use relevant commit messages. Don't forget to keep them clear and concise for easier tracking!
image

For better flexibility, use VS Code instead of GitHub's native code editor. And again, don't forget to use clear and relevant commit messages!
In simple terms, **don't use the same message for every commit**(it's not a good practice...each commit should have a message that reflects the specific changes made).

**Comment by this-is-yaash** [comment]
@Deeptig9138 Additionally, create a new branch from your forker repo and work on it!
Don't use the main branch for working on an issue(again, it's not a good practice)

**Pull request closed without merging**